src/modular/monty_form/inv.rs

```diff
@@ -145,4 +245,323 @@ mod tests {
+
+        let expected_inv = BigUint::from_bytes_be(&hex!("579F198AFBD0EC0921E8626D386C7F8080F8C0668284BE38FE5B9E67AA81A6F637FFEDBB76E9CF68E5E7BA9892D4938DB90906686CEF06A94D16AA9CDA0C0E24DAB9AB72303316266BF4DCF449E00D8F7795819C06CA5921A31B40A2AB1B0D7C264144D0372D59ADD5754FE02B1328E159B4B58767FFB623D1A6B3CC89B3F724A647A9AFCB55ACD02491544849A4603C013C5313DEC80A8AC46C268BA1245BA1B9D05386A560E1CBACE4F7C39873471101C19C6CE07D4CDDE06B1557081F5C838452135A16216285E1AC92A1F30263AC148BBE74A9397514D6B17E7473C703D965EA68054D4AA5AC9967729997A898AFA78C8D418871B30F502F3E01B89F1C3E"));
+        assert_eq!(inverted, expected_inv);
+        let one = int.mulm(inverted, &modulus);
+        assert_eq!(one, BigUint::one());
+    }
+
+    #[test]
+    fn v05_inversion_random_uints_u2048_modulus() {
+        let mut rng = ChaChaRng::from_seed(RANDOM_SEED);
+        let modulus = U2048_05::from_be_hex(concat!(
+            "1fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+            "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+            "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+            "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+            "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+            "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+            "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+            "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+        ));
+
+        let monty_params = DynResidueParams::new(&modulus);
+        for _ in 0..100 {
+            let (r, rm, rmi) = random_invertible_05(monty_params, &mut rng);
+
+            let one_monty = rm * rmi;
+            assert_eq!(one_monty, DynResidue::one(monty_params));
+            assert_eq!(one_monty.retrieve(), U2048_05::ONE);
+
+            let one = rmi.retrieve() * r;
+            let wide_modulus = NonZero_05::new(Into::<U4096_05>::into(&modulus)).unwrap();
+            assert_eq!(one % wide_modulus, U4096_05::ONE);
+
+            let (ri, _) = r.inv_mod(&modulus);
+            assert_eq!(ri, rmi.retrieve());
+        }
+    }
+
+    // Moduli strictly smaller than U2048::MAX / 6 work (i.e. 3 msbs are zero).
+    // In other words, the most significant byte must be smaller than 31 (0x1F).
+    // The test passes when the first bytes is:
+    //      0x1e  00011110,
+    // …and fails when first byte is:
+    //      0x1f  00011111,
+    //
+    // Weirdly enough it seems like U1024 is *not* affected by this problem.
+    // U4096 behaves the same but the threshold is different (see below).
+    #[test]
+    fn v06_inversion_random_uints_u2048_modulus() {
+        let mut rng = ChaChaRng::from_seed(RANDOM_SEED);
+        let modulus = U2048::from_be_hex(concat!(
+            "1fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
```

Comment:
Changing this to `1effffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff` makes the test pass. For this issue it doesn't seem to matter what any other bytes are.